pkg/scheduler/plugins/overcommit/overcommit_test.go

```diff
@@ -0,0 +1,122 @@
+	// pg that requires small resources
+	pg2 := util.BuildPodGroup("pg2", "test-namespace", "c1", 2, nil, schedulingv1.PodGroupPhase(scheduling.PodGroupInqueue))
+	pg2.Spec.MinResources = &hugeResource
+	// pg that no requires resources
+	pg3 := util.BuildPodGroup("pg2", "test-namespace", "c1", 2, nil, schedulingv1.PodGroupPhase(scheduling.PodGroupInqueue))
+
+	queue1 := util.BuildQueue("c1", 1, nil)
+	queue2 := util.BuildQueue("c1", 1, smallResource)
+
+	tests := []struct {
+		uthelper.TestCommonStruct
+		arguments           framework.Arguments
+		expectedEnqueueAble bool
+	}{
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:      "overCommitFactor is more than 0",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg1},
+				Queues:    []*schedulingv1.Queue{queue1},
+				Nodes:     []*v1.Node{n1, n2},
+			},
+			arguments: framework.Arguments{
+				overCommitFactor: 1.2,
+			},
+			expectedEnqueueAble: true,
+		},
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:      "overCommitFactor is less than 0",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg1},
+				Queues:    []*schedulingv1.Queue{queue1},
+				Nodes:     []*v1.Node{n1, n2},
+			},
+			arguments: framework.Arguments{
+				overCommitFactor: 0.8,
+			},
+			expectedEnqueueAble: true,
+		},
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:      "when the required resources of pg are too large",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg2},
+				Queues:    []*schedulingv1.Queue{queue1},
+				Nodes:     []*v1.Node{n1, n2},
+			},
+			arguments: framework.Arguments{
+				overCommitFactor: 1.2,
+			},
+			expectedEnqueueAble: false,
+		},
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:      "when pg does not fill MinResources",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg3},
+				Queues:    []*schedulingv1.Queue{queue2},
+				Nodes:     []*v1.Node{n1, n2},
+			},
+			arguments: framework.Arguments{
+				overCommitFactor: 1.2,
+			},
+			expectedEnqueueAble: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			trueValue := true
+			tiers := []conf.Tier{
+				{
+					Plugins: []conf.PluginOption{
+						{
+							Name:               PluginName,
+							EnabledJobEnqueued: &trueValue,
+							Arguments:          test.arguments,
+						},
+					},
+				},
+			}
+			ssn := test.RegisterSession(tiers, nil)
+			for _, job := range ssn.Jobs {
+				ssn.JobEnqueued(job)
+				isEnqueue := ssn.JobEnqueueable(job)
+				if !reflect.DeepEqual(test.expectedEnqueueAble, isEnqueue) {
+					t.Errorf("case: %s error,  expect %v, but get %v", test.Name, test.expectedEnqueueAble, isEnqueue)
+				}
+			}
+		})
+		test.Close()
```

Comment:
It is better to be in defer function after ssn is registed.

Reply:
done